### PR TITLE
Add introduction page for Shapes Demo

### PR DESCRIPTION
**Pull request by irenebm**
Documentation available [here](https://docs.vulcanexus.org/en/feature-shapes_demo_page/rst/introduction/tools/shapes_demo.html)

### CODE REVIEW

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 1]
```diff
@@ -3,5 +3,136 @@
 ROS 2 Shapes Demo
 =================
 
+*ROS 2 Shapes Demo* is an application in which Publishers and Subscribers are shapes of different colors and sizes moving on a board.
+Each shape refers to its own topic: Square, Triangle or Circle.
+A single instance of the *ROS 2 Shapes Demo* can publish on or subscribe to several topics at a time.
+
+It demonstrates the capabilities of eProsima Fast DDS or as a proof of interoperability with other RTPS-compliant implementations.
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:10]: DDS?

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 2]
```diff
@@ -3,5 +3,136 @@
...
+
+.. figure:: /rst/figures/intro/tools/shapes_demo/mainWindow.png
+   :alt: Main Window
+   :align: center
+
+
+Publishing a Topic
+^^^^^^^^^^^^^^^^^^
+
+The Publish button allow the users to define the Shape (topic) and Quality of Service (QoS) for their publication.
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:33]:
```suggestion
The Publish button allow the users to define the Shape (topic) and Quality of Service for their publication.
```

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 3]
```diff
@@ -3,5 +3,136 @@
...
+
+.. figure:: /rst/figures/intro/tools/shapes_demo/publish.png
+   :scale: 100 %
+   :alt: Publish Window
+   :align: center
+
+There are multiple parameters that the user can define in this menu:
+
+*   Shape: This parameter defines the topic where the publication is going to occur.
+    Three different shapes can be published: ``Square``, ``Circle`` and ``Triangle`` (see Fast DDS Topic Documentation).
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:44]: Maybe a link here (and in every fast dds documentation mention)

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 4]
```diff
@@ -3,5 +3,136 @@
...
+*   Color: The user can define the color of the shape.
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:45]: Add a note that ROS2 does not have keys?

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 5]
```diff
@@ -3,5 +3,136 @@
...
+    This parameter will be used as key; that is, a way to distinguish between multiple instances of the same shape (see Fast DDS Topics with key documentation).
+*   Size: This parameter allows to control the size of the shape.
+    The size can vary between ``1`` and ``99``.
+*   Partition: The user can select different partitions to differentiate groups of publishers and subscribers.
+    The user can select between four partitions (``A``, ``B``, ``C`` and ``D``). Additionally the user can select the ``*`` partition, that will be matched against all other partitions (see Fast DDS Partitions Documentation).
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:50]:
```suggestion
    The user can select *no particion* or any combination between four partitions (``A``, ``B``, ``C`` and ``D``). Additionally the user can select the ``*`` partition, that will be matched against all other partitions (not only the 4 given here) (see Fast DDS Partitions Documentation).
```

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 6]
```diff
@@ -3,5 +3,136 @@
...
+    If the selected ownership is ``EXCLUSIVE`` the publisher will use the Ownership strength value as the strength of its publication.
+    Only the publisher with the highest strength can publish in the same Topic with the same Key (see Fast DDS OwnershipQosPolicy Documentation).
+*   Deadline: The Deadline QoS determines the maximum expected amount of time between samples.
+    When the deadline is missed the application will be notified and a message will be printed on the console (see Fast DDS DeadlineQosPolicy Documentation).
+*   Lifespan: The Lifespan QoS determines the duration while the sample is still valid.
+    When a sample’s lifespan expires, it will be removed from publisher and subscriber histories (see Fast DDS LifespanQosPolicy Documentation).
+
+.. note::
+
+    Using Lifespan QoS will not have any visual effect.
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:72]: Neither Deadline, and most of the times reliability, liveliness neither

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 7]
```diff
@@ -3,5 +3,136 @@
...
+
+*   Liveliness: This QoS policy is applied in the same way as in the publisher except for the Announcement Period, which does not apply for the Subscriber (see Fast DDS LivelinessQosPolicy Documentation).
+*   Time Based Filter: This value can be used by the user to specify the minimum amount of time (in milliseconds) that the subscriber wants between updates (see Fast DDS TimeBasedFilterQosPolicy Documentation).
+*   Content Based Filter: This filter draws a rectangle in the instances window.
+    Only the shapes that are included in this rectangle are accepted while the rest of them are ignored.
+    The user can dynamically resize and move this content filter.
+
+.. note::
+
+    Using Lifespan QoS will not have any visual effect.
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:94]: I would not repeat this

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 8]
```diff
@@ -3,5 +3,136 @@
...
+   :scale: 75 %
+   :alt: Options Window
+   :align: center
+
+*   Transport Protocol: You can select between UDP protocol, TCP LAN Server, TCP WAN Server or TCP Client, and Shared Memory protocol.
+
 .. note::
-    This page is under maintenance and will be updated soon.
+
+    This is under maintenance and will be updated soon.
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:111]: Is it?

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 9]
```diff
@@ -3,5 +3,136 @@
...
+The *ROS 2 Shapes Demo* application allows the user to define Participant policies.
+To see the Options window, please go to Options->Participant Configuration in the main bar.
+The following image shows the Options Menu.
+
+.. figure:: /rst/figures/intro/tools/shapes_demo/participant.png
+   :scale: 75 %
+   :alt: Options Window
+   :align: center
+
+*   Transport Protocol: You can select between UDP protocol, TCP LAN Server, TCP WAN Server or TCP Client, and Shared Memory protocol.
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:107]: I would add every attribute: Same Host delivery, Domain and Statistics

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 10]
```diff
@@ -3,5 +3,136 @@
...
+The user can customize several aspects of Shapes Demo operation:
+
+* Update interval: This value changes the publication period for all the publishers.
+* Speed: This scroll bar allows the user to change how much the Shape moves between two write calls.
+
+Endpoints and Output tabs
+^^^^^^^^^^^^^^^^^^^^^^^^^
+.. note::
+
+    This is under maintenance and will be updated soon.
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:133]: Wouldn't be better to comment this as a todo?

**Review by jparisu — CHANGES_REQUESTED** [review]

File: docs/rst/introduction/tools/shapes_demo.rst (hunk 4 above)
Comment (rsanchez15) [reply docs/rst/introduction/tools/shapes_demo.rst]: I've removed all mentions to topics with keys instead.

File: docs/rst/introduction/tools/shapes_demo.rst (hunk 8 above)
Comment (rsanchez15) [reply docs/rst/introduction/tools/shapes_demo.rst]: This was a note from Irene that I forgot to remove

File: docs/rst/introduction/tools/shapes_demo.rst (hunk 10 above)
Comment (rsanchez15) [reply docs/rst/introduction/tools/shapes_demo.rst]: Same as before

File: docs/rst/introduction/tools/shapes_demo.rst [hunk 11]
```diff
@@ -104,33 +116,48 @@ The following image shows the Options Menu.
...
+    To modify the Domain ID the user needs to stop the participant (thus removing all existing publishers and subscribers) and start a new one with the new Domain ID (see `Fast DDS Domain Documentation <https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/domain/domain.html>`_).
 
-The user can customize several aspects of Shapes Demo operation:
+-   **Statistics**: The user can activate *Fast DDS Statistics module* so different instrumentation data could be
+    collected and analyzed by the *Fast DDS Statistics Backend*, or be represented by *Fast DDS Monitor*.
+    This module requires to have compiled *Fast DDS* with Statistics Module ON (see `Fast DDS Statistics Module Documentation <https://fast-dds.docs.eprosima.com/en/latest/fastdds/statistics/statistics.html>`_).
 
-* Update interval: This value changes the publication period for all the publishers.
-* Speed: This scroll bar allows the user to change how much the Shape moves between two write calls.
+In case that the Participant is already running, it should be stopped in order to change its configuration.
```
Comment (jparisu) [inline docs/rst/introduction/tools/shapes_demo.rst:136]: Add `Use ROS2 Topics` comment.

**Review by jparisu — CHANGES_REQUESTED** [review]
LGTM

**Review by jparisu — APPROVED** [review]
LGTM

**Pull request merged**